apps/docs/guides/building-codemods/codemod2.mdx

```diff
@@ -0,0 +1,177 @@
+---
+title: "Codemod2.0"
+---
+
+This guide shows how you can build, test, and publish a Codemod 2.0 in a few steps. In the following example, we consider a use case where we want to create a codemod to change imports to direct imports:
+
+```tsx
+// BEFORE 
+import {
+  AccessAlarm as AccessAlarmIcon,
+  ThreeDRotation as NiceIcon,
+} from '@mui/icons-material';
+
+//AFTER
+import AccessAlarmIcon from '@mui/icons-material/AccessAlarm';
+import NiceIcon from '@mui/icons-material/ThreeDRotation';
+
+```
+
+### Step 1: Initialize a new codemod.
+
+Run `npx codemod@latest init` in your desired location to store the codemod folder, and select `workflow` as your engine. This sets up the scaffolding for your codemod, making it easy to provide configuration and metadata. While optional for building and running a codemod, this step enhances discoverability and sharing.
+
+![Initializing codemod](/images/guides/codemod2/initializing.png)
+
+### Step 2: Find JS/TS files.
+
+From the previous step, we got a new directory `mui-named-import-to-direct-import` and file inside `src/index.ts`. Let's update this boilerplate file.
+
+We need to get all the files that potentially can have Material UI Icons imports
+
+```tsx
+import type { Api } from '@codemod.com/workflow';
+
+export async function workflow({ files }: Api) {
+  **await files()**
+}
+```
+
+Now let's get Javascript/Typescript context:
+
+```tsx
+import type { Api } from '@codemod.com/workflow';
+
+export async function workflow({ files }: Api) {
+  await files()
+	  **.jsFam()
+}**
+```
+
+### Step 3: Use ast-grep for pattern detection.
+
+Let's use [ast-grep](https://ast-grep.github.io/) which is built-in into our workflow engine
```

Comment:
from here, we should link to workflow engine doc, astGrep section, that explains different ways to use astGrep API in workflow engine... there are different ways we can use this API depending on what the original ast grep codemod looks like... this improvement can happen later once the doc for workflow engine is ready... @arybitskiy , @mohab-sameh

Reply:
Sounds good